Exercise_1.java

```diff
@@ -1,21 +1,49 @@
-class BinarySearch { 
-    // Returns index of x if it is present in arr[l.. r], else return -1 
+class BinarySearch {
+    /**
+     * Returns index of x if it is present in arr[l.. r], else return -1
+     * @param arr integer array
+     * @param l left index in array
+     * @param r right index in array
+     * @param x x is  value to be serched in array
+     * @return integer index of the value x if present in an array else -1
+     */
+    //Time complexity:log(n) where n is array size
+    //Space complexity is:o(1) as I am only using constant space irrespective of my input size.
     int binarySearch(int arr[], int l, int r, int x) 
     { 
-        //Write your code here
+        while(l<=r){
+            int mid=(l+r)/2;

```

Comment:
This will result in integer overflow. We will discuss the issues related to it in Binary Search leg